### ISSUE TITLE
Use Tensor<T>

### ISSUE THREAD

**Issue by franperezlopez** [migueldeicaza/TensorFlowSharp#190]
Hi @migueldeicaza , 

do you have any plans to use [Tensor\<T\>](https://github.com/dotnet/corefxlab) type in your library? IMHO, I'd be great if we could use same Tensor data type. Maybe you could provide some guidelines and the community could help out

**Comment by migueldeicaza** [comment]
I would love to!

**Comment by alexpantyukhin** [comment]
Yes, I think it would be very interesting. Like in the  of [example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/java/src/main/java/org/tensorflow/examples/LabelImage.java#L65) of [Java implementation](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/java/src/main/java/org/tensorflow/Tensor.java#L45).
But for me it's not clear for now how the `TFTensor<T>` would work with `DllImport` ..

**Comment by migueldeicaza** [comment]
The Tensor<T> in Java is a little bit different, it is a strongly-type version of TFTensor as it exists in the C# binding.   But in both cases (Java and my TFTensor) these are merely handles that are used to load data into TensorFlow, they are not generally intended to be used to keep the data around.

I suspect that TFTensor could take Tensor<T> data in the constructor, and also return a Tensor<T> when queried.   Those would involve a copy, it is not the same as what you see in Java.

Another option is to introduce a TFTensor<T> that might help define the base type for the tensor, but it would still operate like what we have today.   Both can be done.
